frameworks/hdfs/src/test/java/com/mesosphere/sdk/hdfs/scheduler/HdfsUserAuthMapperTest.java

```diff
@@ -0,0 +1,111 @@
+        HdfsUserAuthMapper mapper = new HdfsUserAuthMapper(primary, frameworkHost, realm, frameworkUser);
+        mapper.addDefaultUserAuthMapping("data", "node", 1);
+        String expected = "RULE:[2:$1/$2@$0](primary/data-0-node.host@realm)s/.*/user/";
+        Assert.assertEquals(expected, mapper.getUserAuthMappingString());
+    }
+
+    @Test
+    public void testMultipleDefaultUserAuthMappings() {
+        HdfsUserAuthMapper mapper = new HdfsUserAuthMapper(primary, frameworkHost, realm, frameworkUser);
+        mapper.addDefaultUserAuthMapping("data", "node", 2);
+        String expected = "RULE:[2:$1/$2@$0](primary/data-0-node.host@realm)s/.*/user/\nRULE:[2:$1/$2@$0](primary/data-1-node.host@realm)s/.*/user/";
+        Assert.assertEquals(expected, mapper.getUserAuthMappingString());
+    }
+
+    @Test
+    public void testUserAuthMappingSeparation() {
+        HdfsUserAuthMapper mapper = new HdfsUserAuthMapper(primary, frameworkHost, realm, frameworkUser);
+        mapper.addDefaultUserAuthMapping("data", "node", 1);
+        Map<String, String> env = new HashMap<>();
+        String expected = "value";
+        env.put("key", Base64.getEncoder().encodeToString(expected.getBytes(StandardCharsets.UTF_8)));
+        mapper.addUserAuthMappingFromEnv(env, "key");
+        Assert.assertEquals(2, mapper.getUserAuthMappingString().split("\n").length);
+    }
+
+    @Test
+    public void testAuthMapperTemplating() throws Exception {
+        File template = ServiceTestRunner.getDistFile(Main.CORE_SITE_XML);
+        String templateContent = new String(Files.readAllBytes(template.toPath()), StandardCharsets.UTF_8);
+
+        Map<String, String> schedulerEnv =
+                CosmosRenderer.renderSchedulerEnvironment(Collections.emptyMap(), Collections.emptyMap());
+        Map<String, String> taskEnv = new TaskEnvRouter(schedulerEnv).getConfig("ALL");
+        taskEnv.put(EnvConstants.FRAMEWORK_HOST_TASKENV, "hdfs.on.some.cluster");
+        taskEnv.put("MESOS_SANDBOX", "/mnt/sandbox");
+        taskEnv.put(Main.SERVICE_ZK_ROOT_TASKENV, "/dcos-service-path__to__hdfs");
+        taskEnv.put("SECURITY_KERBEROS_ENABLED", "true");
+        taskEnv.put("SECURITY_KERBEROS_PRIMARY", primary);
+        taskEnv.put("SECURITY_KERBEROS_REALM", realm);
+        taskEnv.put("SECURITY_KERBEROS_PRIMARY_HTTP", "primHttp");
+        taskEnv.put("SCHEDULER_API_HOSTNAME", "schedulerHostname");
+
+        HdfsUserAuthMapper mapper = new HdfsUserAuthMapper(primary, frameworkHost, realm, frameworkUser);
+        mapper.addDefaultUserAuthMapping("data", "node", 2);
+
+        taskEnv.put("DECODED_AUTH_TO_LOCAL", mapper.getUserAuthMappingString());
+
+        String filledTemplate = TemplateUtils.renderMustacheThrowIfMissing(template.getName(), templateContent, taskEnv);
+        Assert.assertEquals(findAuthMappings(filledTemplate), mapper.getUserAuthMappingString());
+    }
+
+    private static String findAuthMappings(String config) throws IOException {
```

Comment:
Although I don't have a problem with handling this here -- it's definitely faster, but we could check this here instead: https://github.com/mesosphere/dcos-commons/blob/master/frameworks/hdfs/tests/test_sanity.py#L61. Would it be possible to use an XML-parser in the test instead? Not critical for this PR though. An alternative is would be to not use the full template in this test.

Reply:
Yes, indeed it would be nice to use a proper XML parsing tool. Unfortunately, I haven't found a good one now. Maybe someone with more Java experience might know a proper solution.